### PR TITLE
Add tests for filters srs 03

### PR DESCRIPTION
**Pull request by eCzerniak**
Adding tests for filters. 
Adding `data-testid` to elements.

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-7465 SRS_03 - Add tests for filters in SRS](https://app.clickup.com/t/2549495/SALEOR-7465)

File: cypress/elements/navigation.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const NAVIGATION = {
+  categoriesList: "[data-testid*='categoriesList']",
```
Comment (karola312) [inline cypress/elements/navigation.ts:2]: Wouldn't be better to name it the categoryListButtons? And the same with other elements?

File: cypress/support/shared.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { SHARED } from "cypress/elements/shared";
+
+export function waitForProgressBarToNotBeVisible() {
+  cy.get(SHARED.spinner).should("not.exist");
+}
+
+export function filterProducts(firstElement, secondElement) {
```
Comment (karola312) [inline cypress/support/shared.ts:7]: Variables firstElement, secondElement should be better named. Maybe filterProductsBy, and selectedFilter

File: cypress/e2e/filteringProducts.cy.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+describe("Filter products", () => {
+  beforeEach(() => {
+    cy.visit("/");
+  });
+
+  it("should filter products by variant attribute SRS_0306", () => {
+    waitForProgressBarToNotBeVisible();
+    filterProducts(NAVIGATION.categoriesList, CATEGORY.categoryTitle);
+    cy.get(FILTERS.filtersMenu).first().click();
+    filterProducts(FILTERS.filterList, FILTERS.filterPill);
```
Comment (karola312) [inline cypress/e2e/filteringProducts.cy.ts:15]: Don't you want to check here if the graphql request for filtered products was sent and if all products from the response are displayed?

Comment (eCzerniak) [reply cypress/e2e/filteringProducts.cy.ts:15]: I was thinking about that and it was little bit hard for me in this moment and we can do this if you would like to help me with that ;).

File: cypress/support/shared.ts (hunk 2 above)
Comment (michalina-graczyk) [reply cypress/support/shared.ts:7]: yup, we should avoid using names which doesn't describe what the first element is

**Review by szczecha — APPROVED** [review]

**Review by michalina-graczyk — APPROVED** [review]

**Pull request merged**